containers/src/Data/IntMap/Internal.hs

```diff
@@ -3297,13 +3322,17 @@ INSTANCE_TYPEABLE1(IntMap)
   Link
 --------------------------------------------------------------------}
 link :: Prefix -> IntMap a -> Prefix -> IntMap a -> IntMap a
-link p1 t1 p2 t2
+link p1 t1 p2 t2 = linkWithMask (branchMask p1 p2) p1 t1 p2 t2
+{-# INLINE link #-}
+
+-- `linkWithMask` is useful when the `branchMask` has already been computed
+linkWithMask :: Mask -> Prefix -> IntMap a -> Prefix -> IntMap a -> IntMap a
+linkWithMask m p1 t1 p2 t2
   | zero p1 m = Bin p m t1 t2
   | otherwise = Bin p m t2 t1
   where
```

Comment:
It won't matter, thanks to inlining, but `linkWithMask` doesn't use `p2`, so it shouldn't take it.

Reply:
FWIW, my motivation was to make it easy to switch between `lift` and `liftWithMask`. But of course we may later end up in a situation where `p2` is not readily available... Choices, choices...

Reply:
done